bigtable/client/detail/conjunction.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_DETAIL_CONJUNCTION_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_DETAIL_CONJUNCTION_H_
+
+#include <bigtable/client/version.h>
+#include <type_traits>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace detail {
```

Comment:
Are we switching from "detail" to "internal" at some point?

Reply:
Yes.  Probably after I close out my current branches.